Adding new vault_identity_entity_alias causes runtime panic

```
panic: runtime error: invalid memory address or nil pointer dereference
2019-08-29T15:14:42.161+0200 [DEBUG] plugin.terraform-provider-vault_v2.1.0_x4: [signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x1b9db64]
2019-08-29T15:14:42.161+0200 [DEBUG] plugin.terraform-provider-vault_v2.1.0_x4:
2019-08-29T15:14:42.161+0200 [DEBUG] plugin.terraform-provider-vault_v2.1.0_x4: goroutine 45 [running]:
2019-08-29T15:14:42.161+0200 [DEBUG] plugin.terraform-provider-vault_v2.1.0_x4: github.com/terraform-providers/terraform-provider-vault/vault.identityEntityAliasCreate(0xc00015b5e0, 0x1eb1bc0, 0xc00015a700, 0x2, 0x2c5fb20)
```

### Terraform Version
I have tried all possible combinations between `Terraform v0.12.3`, `Terraform v0.12.7`, `provider.vault v2.1.0` and `provider.vault v2.2.0`

### Affected Resource(s)
- vault_identity_entity_alias

### Terraform Configuration Files
Data has been slightly anonymized to get rid of server and application names.
We manage around 45-50 applications in terraform:
<details><summary>Setup</summary>

We are using this module (`maint.tf`)

```hcl
resource "vault_identity_entity" "application_identity" {
    name      = "${var.application_name}"
    policies  = ["application"]
}

resource "vault_identity_entity_alias" "aliases" {
    count           = length(var.servers)
    name            = "${var.deployment_type}-${var.servers[count.index]}"
    mount_accessor  = "${var.vault_accessor_id}"
    canonical_id    = "${vault_identity_entity.application_identity.id}"
}

resource "vault_mount" "transit" {
    count = var.has_transit_engine ? 1 : 0
    path = "company/transit/apps/${var.application_name}"
    type = "transit"
}
```
To define identities for applications with aliases to a certificate authority which looks like this: 
```hcl
resource "vault_auth_backend" "applications_certificate_auth" {
    path = "company/cert/applications"
    type = "cert"
}
resource "vault_cert_auth_backend_role" "applications_certificate_auth_cert" {
    name          = "CAName"
    certificate   = "${file("./path/to/cert.pem")}"
    backend       = "${vault_auth_backend.applications_certificate_auth.path}"
    policies      = []
}
```
</details>

I was trying to add a new server (`server13`) to one application. No other changes. This was the diff on our config:
old: 
```hcl
module "application_s" {
    source = "./modules/application"
    application_name = "application_s"
    deployment_type = "application-staging"
    servers = ["server1", "server2", "server3", "server4", "server5", "server6", "server8", "server9", "server10", "server11", "server12"]
    vault_accessor_id = "${module.setup.applications_certificate_accessor}"
    has_transit_engine = true
}
```
new:
```hcl
module "application_s" {
    source = "./modules/application"
    application_name = "application_s"
    deployment_type = "application-staging"
    servers = ["server1", "server2", "server3", "server4", "server5", "server6", "server8", "server9", "server10", "server11", "server12", "server13"]
    vault_accessor_id = "${module.setup.applications_certificate_accessor}"
    has_transit_engine = true
}
```
### Panic Output

I also removed application names and servernames from this gist, it is also not the full debug log as that was a huge log and purging it of information was a lot of effort. If you need the entire log, or a bigger part of the log let me know and i'll take the time to generate it.
https://gist.github.com/JeroenBrinkman/0efb73725a1774af38e5bee5eafbb2ed

### Expected Behavior
I expected the planned alias to be added to vault

### Actual Behavior
Vault crashes with a runtime panic

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply` old config to empty vault and no terraform state (dry run)
2. `terraform apply` extra server
3. planned changes are: 
```
  # module.application_s.vault_identity_entity_alias.aliases[11] will be created
  + resource "vault_identity_entity_alias" "aliases" {
      + canonical_id   = "d957a64e-1980-f502-9bdc-e5bc88b8329e"
      + id             = (known after apply)
      + mount_accessor = "auth_cert_b4e16cb9"
      + name           = "application-staging-server13"
    }
```
4. confirm with `yes`
5. crash
6. No new alias and a broken state when combined with other changes that do apply succesfully

### Important Factoids
- Vault version `1.2.2` running in HA mode
- A (assumed) fault in our config causes terraform to recreate the kv2 store every run
- The issue is 100% reproducible on this vault instance however trying the same kind of change on a different instance with about the same number of apps with different names worked perfectly fine. The other instance has less servers however, so in total will contain less aliases.

### References
Not that i know about

--- a/vault/resource_identity_entity_alias.go
+++ b/vault/resource_identity_entity_alias.go
@@ -63,9 +63,22 @@ func identityEntityAliasCreate(d *schema.ResourceData, meta interface{}) error {
 	if err != nil {
 		return fmt.Errorf("error writing IdentityEntityAlias to %q: %s", name, err)
 	}
-	log.Printf("[DEBUG] Wrote IdentityEntityAlias %q", name)
 
-	d.SetId(resp.Data["id"].(string))
+	if resp != nil {
+		d.SetId(resp.Data["id"].(string))
+	} else {
+		// A nil response is possible if the alias already exists and TF doesn't know about it.
+		// But we still need and don't have the alias ID, so a search is used to find it.
+		log.Printf("[DEBUG] Entity alias ID is unknown. Reading aliases for entity %q", canonicalID)
+
+		id, err := findAliasID(client, canonicalID, name, mountAccessor)
+		if err != nil {
+			return fmt.Errorf("error writing IdentityEntityAlias to %q: %s", name, err)
+		}
+		d.SetId(id)
+	}
+
+	log.Printf("[DEBUG] Wrote IdentityEntityAlias %q", name)
 
 	return identityEntityAliasRead(d, meta)
 }
@@ -177,4 +190,23 @@ func identityEntityAliasNamePath(name string) string {
 
 func identityEntityAliasIDPath(id string) string {
 	return fmt.Sprintf("%s/id/%s", identityEntityAliasPath, id)
+}
+
+func findAliasID(client *api.Client, canonicalID, name, mountAccessor string) (string, error) {
+	path := identityEntityIDPath(canonicalID)
+
+	resp, err := client.Logical().Read(path)
+	if err != nil {
+		return "", fmt.Errorf("error reading entity aliases: %s", err)
+	}
+
+	aliases := resp.Data["aliases"].([]interface{})
+	for _, aliasRaw := range aliases {
+		alias := aliasRaw.(map[string]interface{})
+		if alias["name"] == name && alias["mount_accessor"] == mountAccessor {
+			return alias["id"].(string), nil
+		}
+	}
+
+	return "", fmt.Errorf("unable to determine alias ID. canonical ID: %q  name: %q  mountAccessor: %q", canonicalID, name, mountAccessor)
 }

--- a/vault/resource_identity_entity_alias.go
+++ b/vault/resource_identity_entity_alias.go
@@ -64,21 +64,19 @@ func identityEntityAliasCreate(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf("error writing IdentityEntityAlias to %q: %s", name, err)
 	}
 
-	if resp != nil {
-		d.SetId(resp.Data["id"].(string))
-	} else {
-		// A nil response is possible if the alias already exists and TF doesn't know about it.
-		// But we still need and don't have the alias ID, so a search is used to find it.
-		log.Printf("[DEBUG] Entity alias ID is unknown. Reading aliases for entity %q", canonicalID)
+	if resp == nil {
+		aliasIDMsg := "Unable to determine alias id."
 
-		id, err := findAliasID(client, canonicalID, name, mountAccessor)
-		if err != nil {
-			return fmt.Errorf("error writing IdentityEntityAlias to %q: %s", name, err)
+		if aliasID, err := findAliasID(client, canonicalID, name, mountAccessor); err == nil {
+			aliasIDMsg = fmt.Sprintf("Alias resource ID %q may be imported.", aliasID)
 		}
-		d.SetId(id)
+
+		return fmt.Errorf("IdentityEntityAlias %q already exists. %s", name, aliasIDMsg)
 	}
 
 	log.Printf("[DEBUG] Wrote IdentityEntityAlias %q", name)
+
+	d.SetId(resp.Data["id"].(string))
 
 	return identityEntityAliasRead(d, meta)
 }
@@ -200,11 +198,13 @@ func findAliasID(client *api.Client, canonicalID, name, mountAccessor string) (s
 		return "", fmt.Errorf("error reading entity aliases: %s", err)
 	}
 
-	aliases := resp.Data["aliases"].([]interface{})
-	for _, aliasRaw := range aliases {
-		alias := aliasRaw.(map[string]interface{})
-		if alias["name"] == name && alias["mount_accessor"] == mountAccessor {
-			return alias["id"].(string), nil
+	if resp != nil {
+		aliases := resp.Data["aliases"].([]interface{})
+		for _, aliasRaw := range aliases {
+			alias := aliasRaw.(map[string]interface{})
+			if alias["name"] == name && alias["mount_accessor"] == mountAccessor {
+				return alias["id"].(string), nil
+			}
 		}
 	}
 

--- a/vault/resource_identity_entity_alias_test.go
+++ b/vault/resource_identity_entity_alias_test.go
@@ -73,5 +73,14 @@ resource "vault_identity_entity_alias" "entity-alias" {
   name = "%s"
   mount_accessor = "${vault_auth_backend.githubA.accessor}"
   canonical_id = "${vault_identity_entity.entity.id}"
-}`, entityName, entityName, entityName, entityName)
+}
+
+// This duplicate alias tests the provider's handling of aliases that already exist but aren't
+// known to the provider.
+resource "vault_identity_entity_alias" "entity-alias-dupe" {
+  name = "%s"
+  mount_accessor = "${vault_auth_backend.githubA.accessor}"
+  canonical_id = "${vault_identity_entity.entity.id}"
+}
+`, entityName, entityName, entityName, entityName, entityName)
 }

--- a/vault/resource_identity_entity_alias_test.go
+++ b/vault/resource_identity_entity_alias_test.go
@@ -2,6 +2,7 @@ package vault
 
 import (
 	"fmt"
+	"regexp"
 	"testing"
 
 	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
@@ -23,12 +24,16 @@ func TestAccIdentityEntityAlias(t *testing.T) {
 		CheckDestroy: testAccCheckIdentityEntityAliasDestroy,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccIdentityEntityAliasConfig(entity),
+				Config: testAccIdentityEntityAliasConfig(entity, false),
 				Check: resource.ComposeTestCheckFunc(
 					resource.TestCheckResourceAttr(nameEntityAlias, "name", entity),
 					resource.TestCheckResourceAttrPair(nameEntityAlias, "canonical_id", nameEntity, "id"),
 					resource.TestCheckResourceAttrPair(nameEntityAlias, "mount_accessor", nameGithubA, "accessor"),
 				),
+			},
+			{
+				Config:      testAccIdentityEntityAliasConfig(entity, true),
+				ExpectError: regexp.MustCompile(`IdentityEntityAlias.*already exists.*may be imported`),
 			},
 		},
 	})
@@ -52,8 +57,8 @@ func testAccCheckIdentityEntityAliasDestroy(s *terraform.State) error {
 	return nil
 }
 
-func testAccIdentityEntityAliasConfig(entityName string) string {
-	return fmt.Sprintf(`
+func testAccIdentityEntityAliasConfig(entityName string, dupeAlias bool) string {
+	ret := fmt.Sprintf(`
 resource "vault_identity_entity" "entity" {
   name = "%s"
   policies = ["test"]
@@ -74,13 +79,19 @@ resource "vault_identity_entity_alias" "entity-alias" {
   mount_accessor = "${vault_auth_backend.githubA.accessor}"
   canonical_id = "${vault_identity_entity.entity.id}"
 }
+`, entityName, entityName, entityName, entityName)
 
-// This duplicate alias tests the provider's handling of aliases that already exist but aren't
-// known to the provider.
+	// This duplicate alias tests the provider's handling of aliases that already exist but aren't
+	// known to the provider.
+	if dupeAlias {
+		ret += fmt.Sprintf(`
 resource "vault_identity_entity_alias" "entity-alias-dupe" {
   name = "%s"
   mount_accessor = "${vault_auth_backend.githubA.accessor}"
   canonical_id = "${vault_identity_entity.entity.id}"
 }
-`, entityName, entityName, entityName, entityName, entityName)
+`, entityName)
+	}
+
+	return ret
 }

